lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2768,6 +2770,34 @@ public function createEntity($className, array $data, &$hints = array())
     }
 
     /**
+     * @param array $data
+     * @param ClassMetadata $class
+     */
+    private function removeNullableEmbeddedReferences(array &$data, ClassMetadata $class)
```

Comment:
@Ocramius I've tried to keep the things simple here, but if we have a huge amount of data or a large result set I think we might have a little delay to process. I just don't have any idea to improve this =/